src/codegen/compile.ml

```diff
@@ -10082,9 +10082,15 @@ and compile_lit_pat env l =
     todo_trap env "compile_lit_pat" (Arrange_ir.lit l)
 
 and fill_pat env ae pat : patternCode =
+  let irrefutable_nonbinding p =
+    Ir_utils.is_irrefutable pat && Freevars.(M.is_empty (pat p)) in
```

Comment:
I'd have put this in some module near the Ir definition (it's not really backend specific), but not important.

I think the backend has special code for _ elsewhere, which could be also be generalized to this predicate. It's in a way a ”all forms of wild patterns”.

Reply:
I moved it in `Ir_utils`, feel free to use elsewhere!